Source/AwsCommonRuntimeKit/http/HttpStreamCalbackCore.swift

```diff
@@ -52,11 +52,9 @@ private func onResponseHeaders(stream: UnsafeMutablePointer<aws_http_stream>?,
     var headers = [HttpHeader]()
 
     for cHeader in UnsafeBufferPointer(start: headerArray, count: headersCount) {
-        if let name = cHeader.name.toString(),
-           let value = cHeader.value.toString() {
-            let swiftHeader = HttpHeader(name: name, value: value)
-            headers.append(swiftHeader)
-        }
+        let name = cHeader.name.toString()
+        let value = cHeader.value.toString()
+        headers.append(HttpHeader(name: name, value: value))
```

Comment:
we're not even copying the `compression` member

possible solutions:
- Give HttpHeader a private init that takes the C struct (and remembers to copy the compression)
- remove the `compression` member altogether until we're really ready to expose it to users (we haven't exposed it to any other language bindings ... it's a feature we're still kind of figuring out...)

Reply:
I have already refactored the headers logic in #108. We can refactor it further as part of that PR.